### DISCUSSION TITLE
How to add SSR to an existing application?

### DISCUSSION THREAD

**Discussion by lopesrbruno** [TanStack/router#1878] [Q&A]
Hey there!

First of all, I know that TanStack Start is in development, but since I don't know the ETA for Start, what do I need to add SSR capabilities to an application already in production that is using TanStack Router and Query?

**Comment by SeanCassiere** [comment]
Tbh, for SSR, Start is pretty stable. We are using it for TanStack.com. 

If you want to go down the path of manually adding in SSR, it isn't the craziest thing, since all you need to do is spin up a server to handle requests (express, hono, etc... take your pick). Then it's a matter of writing an entry file and handler to handle running React on the server, something like [this](https://github.com/TanStack/router/blob/main/examples/react/basic-ssr-file-based/src/entry-server.tsx).

I'd honestly, say take a look at the `examples/react/start-basic-react-query` example first, before you go about doing manual SSR.
